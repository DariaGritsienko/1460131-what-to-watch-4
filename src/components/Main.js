import React from 'react';
import PropTypes from 'prop-types';
import GenresList from './GenresList';
import ShowMore from './ShowMore';
import VideoPlayerFull from './VideoPlayerFull';
import {openVideoPlayerAction} from '../reducer';

export default class Main extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      page: 0,
      isButtonShowMoreActive: false,
    };
  }

  onButtonPlayClick() {
    this.props.store.dispatch(openVideoPlayerAction(true));
    this.setState({isVideoPlayerOpen: true});
  }

  onCloseVideoPlayer(isPlay) {
    this.setState({isVideoPlayerOpen: isPlay});
    this.props.store.dispatch(openVideoPlayerAction(isPlay));
  }

  onButtonClickMore(page) {
    this.setState({page});
  }

  onButtonShowMoreActive(isActive) {
    this.setState({isButtonShowMoreActive: isActive});
  }

  render() {
    const {filmsData, onFilmsTitleClick, store} = this.props;
    const cardTitle = filmsData[0].title;
    if (!store) {
      return null;
    }

    return (
      <>
        {
          store.getState().isVideoPlayerOpen
          && <VideoPlayerFull
            film={filmsData[0]}
            onCloseVideoPlayer={(isPlay) => this.onCloseVideoPlayer(isPlay)}
            store={store}
          />
          || <div className="main">
            <section className="movie-card">
              <div className="movie-card__bg">
                <img src="img/bg-the-grand-budapest-hotel.jpg" alt="The Grand Budapest Hotel" />
              </div>

              <h1 className="visually-hidden">WTW</h1>

              <header className="page-header movie-card__head">
                <div className="logo">
                  <a className="logo__link">
                    <span className="logo__letter logo__letter--1">W</span>
                    <span className="logo__letter logo__letter--2">T</span>
                    <span className="logo__letter logo__letter--3">W</span>
                  </a>
                </div>

                <div className="user-block">
                  <div className="user-block__avatar">
                    <img src="img/avatar.jpg" alt="User avatar" width="63" height="63" />
                  </div>
                </div>
              </header>

              <div className="movie-card__wrap">
                <div className="movie-card__info">
                  <div className="movie-card__poster">
                    <img src="img/the-grand-budapest-hotel-poster.jpg" alt="The Grand Budapest Hotel poster" width="218" height="327" />
                  </div>
                  <div className="movie-card__desc">
                    <h2 className="movie-card__title">{cardTitle}</h2>
                    <p className="movie-card__meta">
                      <span className="movie-card__genre">{filmsData[0].genre}</span>
                      <span className="movie-card__year">{filmsData[0].year}</span>
                    </p>

                    <div className="movie-card__buttons">
                      <button className="btn btn--play movie-card__button" onClick={() => this.onButtonPlayClick()} type="button">
                        <svg viewBox="0 0 19 19" width="19" height="19">
                          <use xlinkHref="#play-s"></use>
                        </svg>
                        <span>Play</span>
                      </button>
                      <button className="btn btn--list movie-card__button" type="button">
                        <svg viewBox="0 0 19 20" width="19" height="20">
                          <use xlinkHref="#add"></use>
                        </svg>
                        <span>My list</span>
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </section>

            <div className="page-content">
              <section className="catalog">
                <h2 className="catalog__title visually-hidden">Catalog</h2>
                <GenresList store={store} page={this.state.page} onButtonShowMoreActive={(isActive) => this.onButtonShowMoreActive(isActive)} filmsData={filmsData} onFilmsTitleClick={onFilmsTitleClick}/>
                <div className="catalog__more">
                  <ShowMore store={store} onButtonClickMore={(page) => this.onButtonClickMore(page)} isButtonShowMoreActive={this.state.isButtonShowMoreActive}/>
                </div>
              </section>

              <footer className="page-footer">
                <div className="logo">
                  <a className="logo__link logo__link--light">
                    <span className="logo__letter logo__letter--1">W</span>
                    <span className="logo__letter logo__letter--2">T</span>
                    <span className="logo__letter logo__letter--3">W</span>
                  </a>
                </div>

                <div className="copyright">
                  <p>Â© 2019 What to watch Ltd.</p>
                </div>
              </footer>
            </div>
          </div>
        }
    </>
    );
  }

}


Main.propTypes = {
  filmsData: PropTypes.array.isRequired,
  store: PropTypes.object,
  onFilmsTitleClick: PropTypes.func
};
