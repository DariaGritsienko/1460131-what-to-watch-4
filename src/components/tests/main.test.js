import React from "react";
import renderer from "react-test-renderer";
import Main from '../Main';

const filmsData = [
  {
    title: `Some film1`,
    year: `1234`,
    genre: `Comedy`,
    about: {
      poster: `img/the-grand-budapest-hotel-poster.jpg`,
      cover: `img/bg-the-grand-budapest-hotel.jpg`,
      score: `8,9`,
      level: `Very good`,
      rating: `240 ratings`,
      text: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
      moreText: `Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
      director: `Wes Andreson`,
      starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`, `Saoirse Ronan`, `Tony Revoloru`, `Tilda Swinton`, `Tom Wilkinson`, `Owen Wilkinson`, `Adrien Brody`, `Ralph Fiennes`, `Jeff Goldblum`],
      trailer: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
      runTime: `1h 39m`,
      reviewer: [
        {
          reviewerName: `Kate Muir`,
          reviewDate: `2016-12-24`,
          reviewText: [`Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`],
          score: `8,9`,
        },
        {
          reviewerName: `Bill Goodykoontz`,
          reviewDate: `2015-11-18`,
          reviewText: [`Anderson's films are too precious for some, but for those of us willing to lose ourselves in them, they're a delight. "The Grand Budapest Hotel" is no different, except that he has added a hint of gravitas to the mix, improving the recipe.`],
          score: `8,0`,
        },
      ]
    },
  },
];

describe(`Foo`, () => {
  it(`<Main /> should render main page`, () => {
    const tree = renderer.create(
        <Main
          filmsData={filmsData}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
