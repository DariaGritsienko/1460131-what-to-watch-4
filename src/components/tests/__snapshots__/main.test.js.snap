// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Foo <Main /> should render main page 1`] = `null`;
